src/dotnet/Program.cs

```diff
@@ -178,6 +182,9 @@ private static void ConfigureDotNetForFirstTimeUse()
                             environmentProvider);
 
                         dotnetConfigurer.Configure();
+
+                        if (!File.Exists(Telemetry.TelemetrySentinel))
```

Comment:
For first run experience, I went to quite some trouble to isolate this logic of manipulating files, etc. Can you take a look at that and use something similar?

We should probably create an abstraction ITelemetrySentinel (like we have for the NuGet cache) and then we can unit test that.